### PR TITLE
173 Improvements 2.0 (LESS FUCK VIVADEL)

### PR DESCRIPTION
**Pull request by Toxici11i**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
tweak: 173 can now see farther and has less retarded nightvision
fix: 173 no longer can be "seen" in darkness
fix: 173 can now attack simplemobs
tweak: 173 can also attack anything that has the living flag
/:cl:

## About The Pull Request
Vivadel closed the other one, even though this is a fix/tweak PR with positive shit.
fuck niggerdel.

## Why It's Good For The Game
literally fixes 173 attacks, movement etc.

### CODE REVIEW

**Comment by Toxici11i** [comment]
travis is failing still due to a linter issue with grape blast, not my issue.

**Comment by Toxici11i** [comment]
As described in the old PR, this is the full list of shit changed
```
tweak: better ways of checking for attacks, blinking and some more feedback to the player
tweak: blink code improved
tweak: SCP-173 starts making people blink further away
tweak: 173 can now attack simplemobs
tweak: Statuecode blind spell now cannot blind the user
tweak: 173 can now see further than humans, through walls and has native nightvision (NV SPELL BEING KEPT JUST INCASE)
tweak: 173 can no longer speak
fix: 173 should no longer blind himself
add: 173 now has functioning AI
tweak: 173 now has a much higher crushforce
```

**Comment by Toxici11i** [comment]
>refuses to review content
maintainer moments

File: code/modules/mob/living/simple_animal/hostile/statue.dm [hunk 1]
```diff
@@ -190,8 +190,10 @@
 /obj/effect/proc_holder/spell/aoe_turf/blindness/cast(list/targets,mob/user = usr)
 	for(var/mob/living/L in GLOB.alive_mob_list)
 		var/turf/T = get_turf(L.loc)
-		if(T && (T in targets))
+		if(T && T in targets && T != usr)
```
Comment (yoyobatty) [inline code/modules/mob/living/simple_animal/hostile/statue.dm]: use user not usr

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 2]
```diff
@@ -10,21 +10,36 @@
...
+	see_in_dark = 13
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	vision_range = 12
+	aggro_vision_range = 12
+	sight = SEE_SELF|SEE_MOBS|SEE_OBJS|SEE_TURFS
+
+	search_objects = 1 // So that it can see through walls
+
+/mob/living/simple_animal/hostile/scp_173/say(message, bubble_type, var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
+	return 0
```
Comment (yoyobatty) [inline hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: return FALSE

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 1 above)
Comment (yoyobatty) [reply code/modules/mob/living/simple_animal/hostile/statue.dm]: Also why are you checking if the usr is not the TURF? And why every living mob? Shouldn't you just get mobs in range instead, or does it blind everyone?

**Comment by AnthonyJNgo** [comment]
I can’t believe you actually sifted through that, bless your soul Batty.

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 1 above)
Comment (Toxici11i) [reply code/modules/mob/living/simple_animal/hostile/statue.dm]: it blinds everyone in a certain distance to stop people from stacking 4-5 people around 173 (e.g 173's ability to flicker the lights as in CB)

File: hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 2 above)
Comment (Toxici11i) [reply hippiestation/code/modules/SCPmodular/scps/euclid/SCP-173.dm]: fair

**Comment by Toxici11i** [comment]
the original commits were tested.

File: code/modules/mob/living/simple_animal/hostile/statue.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/statue.dm]: > it blinds everyone in a certain distance 

so why do you need to loop through `GLOB.alive_mob_list` for this

Comment (Toxici11i) [reply code/modules/mob/living/simple_animal/hostile/statue.dm]: the for loop wasnt even my code, the only thing that i changed was the T != usr, ask the original coder.

Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/statue.dm]: Oh wait, I kinda see now. It'd stop people in lockers/mechs from cheesing it.

**Comment by Toxici11i** [comment]
@AnthonyJNgo exist

**Review by Neotwr — APPROVED** [review]

**Pull request merged**